### ISSUE TITLE
[ClayCss] Enable utilities classes for a better development experience 

### ISSUE THREAD

**Issue by eduardoallegrini** [liferay/clay#2993]
Hi guys, recently I've been working on the Questions project https://issues.liferay.com/browse/LPS-109146 During this project I had to restyle the app and I had to choose between custom classes and Bootstrap's utility classes.

I started with the custom classes and then I switched to ultilities because we had a short deadline and the project was oriented towards a POC. The difference of working with utilities classes is really relevant, the markup creation is so fluid and fast, the responsiveness is ready to be defined and most of the styles that we need already exists.

In conclusion I think we should allow and recommend the use of the utilities classes in our projects.

The problem is that at the moment (in Bootstrap 4) most of the classes have the important attribute, while In Bootstrap 5 is optional (thanks to a Marcos' request https://github.com/twbs/bootstrap/pull/29625)

So my idea is to either to backport their update from Bootstrap 5 or create our own utilities classes in Clay as we did as a test for the spacing ones https://github.com/liferay/clay/issues/2725

**Comment by pat270** [comment]
I don't see the problem with `!important` if we are using this for POC's. If you want utilities without `!important` we can try adding it and see how much weight it adds to the final css file. I think it should gzip pretty well, but who knows. What we can't do is backport from Bootstrap 5 (unless we are ok with breaking other people's implementations) and set the flag not to use `!important` since we've already released any don't know who is relying on the `!important` flag.

Feel free to send a PR.

**Comment by pat270** [comment]
Duplicating BS4 utilities without `!important` will add:
|theme|additional size|
|---|---|
|atlas.css|57kb|
|atlas.css.gz|7kb|
|base.css|56kb|
|base.css.gz|7kb|

The classes will be prefixed with `.c-`. Now the question, is the additional weight worth it?

**Comment by jbalsas** [comment]
Is `atlas.css` additional size a grand total of 56+57=113kb, or just 57kb?

It feels a tad bit much right off the door... specially if we can't leverage some improvement to compensate. We should start operating under a budget, to add 56kb here we should cut them elsewhere.

@bryceosterhaus, this ties to part of the conversation about performance and testing that we talked about earlier.

**Comment by marcoscv-work** [comment]
Most used helper classes are related to margin/padding, they are currently ready in Clay.
From my point of view, it does not worth it.

**Comment by pat270** [comment]
`atlas.css` size is 57kb, has nothing to do with `base.css` size.

**Comment by pat270** [comment]
@marcoscv-work I duplicated margin/padding CSS. New numbers:

|theme|size|
|---|---|
|atlas.css|21kb|
|atlas.css.gz|3kb|
|base.css|20kb|
|base.css.gz|3kb|

**Comment by eduardoallegrini** [comment]
Aren't the spacing-utilties already merged in Clay? https://github.com/liferay/clay/blob/master/packages/clay-css/src/scss/components/_utilities.scss#L324

Anyway, when I was working on the [question-web module](https://github.com/liferay-frontend/liferay-portal/tree/master/modules/apps/questions/questions-web) I missed some utility classes like `text-secondary`, `text-primary`, `small`, `d-flex`, `d-block`, `align-items-center`, `justify-content-center`, etc.. So I think we should consider to include [utilities-color](https://getbootstrap.com/docs/4.5/utilities/colors/), [utilities-display](https://getbootstrap.com/docs/4.5/utilities/display/) and [utilities-flex](https://getbootstrap.com/docs/4.5/utilities/flex/)

**Comment by kresimir-coko** [comment]
I know you did some work removing/absorbing Bootstrap recently @pat270, has that maybe accidentally fixed this issue?

**Comment by pat270** [comment]
@kresimir-coko 20kb was too much to justify adding it. Holding off until we definitely need it.
